packages/ampd-sdk/src/grpc/error.rs

```diff
@@ -41,12 +42,15 @@ pub enum AppError {
 
     #[error("invalid byte array")]
     InvalidByteArray,
+
+    #[error("connection to the manager is unavailable")]
+    ConnectionUnavailable,
 }
 
 #[derive(Error, Debug)]
 pub enum Error {
-    #[error("failed to connect to the grpc endpoint")]
-    GrpcConnection(#[from] tonic::transport::Error),
+    #[error("failed to reconnect to the pool")]
+    ConnectionStateReceiveFailed(#[from] watch::error::RecvError),
```

Comment:
is there no relevant information in the RecvError? This will discard the inner error